src/dictionary/flat_file/flat_file_dictionary_handler.c

```diff
@@ -197,6 +197,21 @@ ffdict_find(
 			}
 		}
 
+		case predicate_all_records: {
+			ffdict_cursor_t *ffdict_cursor = (ffdict_cursor_t *) (*cursor);
+
+			if (NULL == ((ff_file_t *) dictionary->instance)->start_of_data) {
+				(*cursor)->status = cs_cursor_uninitialized;
+			}
+			else {
+				ffdict_cursor->current	= ((ff_file_t *) dictionary->instance)->start_of_data;
```

Comment:
Here we should call `ffdict_scan` like we do on line 189 and behave accordingly. This should let us "exit early" in the case that there is no data to be read.